src/Symfony/Component/TypeInfo/Type/CollectionType.php

```diff
@@ -34,6 +34,11 @@ public function __construct(
         private readonly BuiltinType|ObjectType|GenericType $type,
         private readonly bool $isList = false,
     ) {
+        // BC layer to allow invalid list keys
+        if (\func_get_args()[2] ?? false) {
```

Comment:
since it's a constructor, we can use a real argument here, it'd be more explicit

Reply:
By doing that, the following, the list keys validation will be explicit like so:
```php
public function __construct(
    private readonly BuiltinType|ObjectType|GenericType $type,
    private readonly bool $isList = false,

    // BC layer to allow invalid list keys
    // Can be removed as soon as PropertyTypeExtractorInterface::getTypes() is removed (8.0).
    private readonly bool $validateListKeys = true,
) {
}
```
Then, how to remove it in 8.0 without breaking backward compatibility?

Reply:
If allowing invalid keys is an implementation detail of the deprecated PropertyInfo's `Type` that we would like to ultimately get rid of in TypeInfo, we probably want to explicitly deprecate this behavior on its own **after** that the PropertyInfo API is removed anyway. This would allow for smooth upgrades as opposed to silently removing that behavior along with the deprecated PropertyInfo's `Type`.
i.e. make it a real parameter in TypeInfo now as suggested, then deprecate it in 8.1, and finally remove it in 9.0.

Reply:
Or maybe we could introduce the parameter and deprecate it right now with an explicit notice?

Reply:
I'd rather go with the latter one indeed, as `CollectionType` has been released yet.